packages/apputils/src/menufactory.ts

```diff
@@ -58,7 +58,6 @@ export namespace MenuFactory {
   ): Menu {
     const menu = menuFactory(item);
     menu.id = item.id;
-    menu.title.label = item.label ?? capitalize(menu.id);
```

Comment:
Unfortunately this will break the case of non existing menu or submenu; for example in the extension @jupyterlab/git a new menu is added. Without this, the menu will not have any label.

I see two possibilities, either the value is translated when the settings are read or the translator is passed up to here (I would prefer this approach).

Do you want me to take over the modifications on this one?

Reply:
I did not delete this code, just moved it to a factory a level higher up to the `generateMenu` which gets passed as `menuFactory` callback to the menufactory module so this should do the trick, right? I did it this way to minimize changes to public API so we can release it under 3.1 easily.

It currently works fine with master (which has a hard-coded label and uses deprecated `.addMenu()`) - in that case GitMenu does not get passed to `MainMenu.generateMenu`:

![Screenshot from 2021-07-30 10-32-34](https://user-images.githubusercontent.com/5832902/127635254-a9ee17c9-e939-4cce-af10-40f4d985625f.png)

I also checked with your WIP branch:

![Screenshot from 2021-07-30 11-07-22](https://user-images.githubusercontent.com/5832902/127637871-a9784612-c905-4ef6-8d59-94a83dc95b87.png)

And I verified that the GitMenu does get passed to `MainMenu.generateMenu`. So i believe all is good.

Reply:
Ok thanks for trying out on the jupyterlab-git extension.

I thought it was gonna break the context menus that uses that function too.

But actually the translation is done in the menu factory for the context menu too (already forgot my work...):

https://github.com/jupyterlab/jupyterlab/blob/758c00ce586738f4c7ba942416f4bf4762ba120e/packages/application-extension/src/index.tsx#L619

So this is ok for me to translate the label in the menu factory. But I would keep that line for consistent behavior that the menu is set from the data. Otherwise if somebody reuses this one, it may be complicated to find out why the label is not set. 

A comment that the function is not taking care of the translation would probably help too.

Reply:
> But I would keep that line for consistent behavior that the menu is set from the data.

But then it over-writes the translated string as it is *after* the call to `menuFactory` (and has to be after because earlier we do not have a menu to work with).

> Otherwise if somebody reuses this one, it may be complicated to find out why the label is not set.

Yes, this is correct. Would it be sufficient to document that they need to do this in the `menuFactory` callback (which they need to provide anyways).